File: Backend/src/main/java/vn/pawstag/service/AuthService.java
package vn.pawstag.service;

import vn.pawstag.dto.request.ForgotPasswordRequest;
import vn.pawstag.dto.request.GoogleLoginRequest;
import vn.pawstag.dto.request.LoginRequest;
import vn.pawstag.dto.request.RegisterRequest;
import vn.pawstag.dto.request.ResetPasswordRequest;
import vn.pawstag.dto.response.AuthSession;

public interface AuthService {
    AuthSession register(RegisterRequest request);
    AuthSession login(LoginRequest request);

    /** Đăng nhập/đăng ký bằng Google ID token (verify với Google, tìm/link/tạo owner). */
    AuthSession googleLogin(GoogleLoginRequest request);

    /** Gửi mã OTP về email nếu tài khoản tồn tại. Luôn "thành công" ở phía client (chống dò email). */
    int forgotPassword(ForgotPasswordRequest request);

    /** Xác nhận OTP + đặt mật khẩu mới. */
    void resetPassword(ResetPasswordRequest request);
}

File: Frontend/pawtag-web/src/services/auth.service.ts
import api from "@/lib/axios";
import type { User } from "@/types";

export interface AuthData {
  owner: User;
}

export interface ForgotPasswordData {
  resendCooldownSeconds: number;
}

// Backend bọc { success, data, message } → unwrap .data.data
export const authService = {
  login: async (email: string, password: string): Promise<AuthData> =>
    (await api.post("/auth/login", { email, password })).data.data,

  register: async (name: string, email: string, password: string, phone?: string): Promise<AuthData> =>
    (await api.post("/auth/register", { name, email, password, phone })).data.data,

  // Gửi Google ID token (credential) — backend verify với Google rồi set cookie HttpOnly.
  googleLogin: async (credential: string): Promise<AuthData> =>
    (await api.post("/auth/google", { credential })).data.data,

  // Gửi Facebook access token — backend verify với Graph API rồi set cookie HttpOnly.
  facebookLogin: async (accessToken: string): Promise<AuthData> =>
    (await api.post("/auth/facebook", { accessToken })).data.data,

  me: async (): Promise<User> => (await api.get("/owners/me")).data.data,

  logout: () => api.post("/auth/logout").catch(() => {}),

  forgotPassword: async (email: string): Promise<ForgotPasswordData> =>
    (await api.post("/auth/forgot-password", { email })).data.data,

  resetPassword: async (email: string, otp: string, newPassword: string): Promise<void> => {
    await api.post("/auth/reset-password", { email, otp, newPassword });
  },
};
